x-pack/plugins/watcher/server/models/watch_status_model/watch_status_model.test.ts

```diff
@@ -0,0 +1,142 @@
+      expect(() => {
+        // @ts-ignore
+        buildServerWatchStatusModel({});
+      }).toThrow(/must contain an id property/i);
+    });
+
+    // TODO: Remove once all consumers and upstream dependencies are converted to TS.
+    it(`throws an error if no 'watchStatusJson' property in json`, () => {
+      expect(() => {
+        // @ts-ignore
+        buildServerWatchStatusModel({ id: 'test ' });
+      }).toThrow(/must contain a watchStatusJson property/i);
+    });
+
+    it('returns correct object for use by Kibana server', () => {
+      const serverWatchStatusModel = buildServerWatchStatusModel(upstreamJson);
+
+      expect(serverWatchStatusModel.id).toBe(upstreamJson.id);
+      expect(serverWatchStatusModel.watchStatusJson).toEqual(upstreamJson.watchStatusJson);
+      expect(serverWatchStatusModel.isActive).toEqual(true);
+      expect(serverWatchStatusModel.lastChecked).toEqual(
+        moment(upstreamJson.watchStatusJson.last_checked)
+      );
+      expect(serverWatchStatusModel.lastMetCondition).toEqual(
+        moment(upstreamJson.watchStatusJson.last_met_condition)
+      );
+
+      expect(serverWatchStatusModel.actionStatuses!.length).toBe(2);
+
+      expect(serverWatchStatusModel.actionStatuses![0]).toMatchObject({
+        id: 'foo',
+        actionStatusJson: {
+          ack: {
+            state: 'awaits_successful_execution',
+            timestamp: '2015-05-26T18:21:08.630Z',
+          },
+          last_execution: {
+            successful: true,
+            timestamp: '2017-07-05T00:00:00.000Z',
+          },
+        },
+        errors: undefined,
+        lastExecutionSuccessful: true,
+        lastExecutionReason: undefined,
+        lastThrottled: null,
+        lastSuccessfulExecution: null,
+        lastExecutionRawFormat: '2017-07-05T00:00:00.000Z',
+        lastCheckedRawFormat: '2017-03-02T14:25:31.139Z',
+        lastAcknowledged: moment('2015-05-26T18:21:08.630Z'),
+        lastExecution: moment('2017-07-05T00:00:00.000Z'),
+      });
+
+      expect(serverWatchStatusModel.actionStatuses![1]).toMatchObject({
+        id: 'bar',
+        actionStatusJson: {
+          ack: {
+            state: 'awaits_successful_execution',
+            timestamp: '2015-05-26T18:21:08.630Z',
+          },
+          last_execution: {
+            successful: true,
+            timestamp: '2017-07-05T00:00:00.000Z',
+          },
+        },
+        errors: undefined,
+        lastExecutionSuccessful: true,
+        lastExecutionReason: undefined,
+        lastThrottled: null,
+        lastSuccessfulExecution: null,
+        lastExecutionRawFormat: '2017-07-05T00:00:00.000Z',
+        lastCheckedRawFormat: '2017-03-02T14:25:31.139Z',
+        lastAcknowledged: moment('2015-05-26T18:21:08.630Z'),
+        lastExecution: moment('2017-07-05T00:00:00.000Z'),
+      });
+    });
+  });
+
+  describe('buildClientWatchStatusModel', () => {
+    it('returns correct object for use by Kibana client', () => {
```

Comment:
I know the comment and state fields are tested more heavily in `watch_status_model_utils.test`, but, for completeness here, should we also add expect statements for `serverWatchStatusModel.comment` and `serverWatchStatusModel.state`?

Reply:
Done.